crates/kv-store/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+//! # MemKvStore Documentation
+//!
+//! MemKvStore use SSTable as backend. The SSTable (Sorted String Table) is a persistent data structure
+//! used for storing key-value pairs in a sorted manner. This document describes the binary format of
+//! the SSTable.
+//!
+//! ## Overall Structure
+//!
+//! The SSTable consists of the following sections:
+//!
+//! ┌─────────────────────────────────────────────────────────────────────────────────────────────────┐
+//! │ MemKVStore                                                                                      │
+//! │┌ ─ ─ ─ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ┬ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ─ ┬ ─ ─ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ┐│
+//! │  Magic Number │ Schema Version │ Block Chunk   ...  │  Block Chunk    Block Meta │ Meta Offset  │
+//! ││     u32      │       u8       │    bytes    │      │     bytes     │   bytes    │     u32     ││
+//! │ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ │
+//! └─────────────────────────────────────────────────────────────────────────────────────────────────┘
+//!
+//! 1. Magic Number (4 bytes): A fixed value "LORO" to identify the file format.
+//! 2. Schema Version (1 byte): The version of the MemKVStore schema.
+//! 3. Block Chunks: A series of data blocks containing key-value pairs.
+//! 4. Block Meta: Metadata for all blocks, including block offset, the first key of the block, `is_large` flag, and last key
+//!    if not large.
+//! 5. Meta Offset (4 bytes): The offset of the Block Meta section from the beginning of the file.
+//!
+//! ## Block Types
+//!
+//! There are two types of blocks: Normal Blocks and Large Value Blocks.
+//!
+//! ### Normal Block
+//!
+//! Normal blocks store multiple key-value pairs with compressed keys.
+//!
+//! ┌────────────────────────────────────────────────────────────────────────────────────────────┐
+//! │Block                                                                                       │
+//! │┌ ─ ─ ─ ─ ─ ─ ─ ┬ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ─ ┬ ─ ─ ─ ─┌ ─ ─ ─┌ ─ ─ ─ ┬ ─ ─ ─ ─┌ ─ ─ ─ ─ ─ ─ ─     │
+//! │ Key Value Chunk  ...  │Key Value Chunk  offset │ ...  │ offset  kv len │Block Checksum│    │
+//! ││     bytes     │      │     bytes     │  u16   │      │  u16  │  u16   │     u32           │
+//! │ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ─ ┘─ ─ ─ ─ ─ ─ ─ ┘    │
+//! └────────────────────────────────────────────────────────────────────────────────────────────┘
+//!
+//! Each Key Value Chunk is encoded as follows:
+//!
+//! ┌─────────────────────────────────────────────────────┐
+//! │  Key Value Chunk                                    │
```

Comment:
Does it include the checksum?

Reply:
Unnecessary, I think the block checksum is sufficient.

Reply:
Oh, yes. This is a chunk